lib/Controls.tsx

```diff
@@ -98,6 +191,159 @@ export const Controls: React.FC<ControlsProps> = ({
+   *  where the duration is longer than the end of the range.
+   *
+   * When computing progress, if the duration is Infinity (live playback), we
+   * use the total buffered time as a (temporary) duration.
+   */
+  const [progress, setProgress] = useState({
+    playedFraction: 0,
+    bufferedFraction: 0,
+    counter: '',
+  })
+  useEffect(() => {
+    if (videoProperties === undefined) {
+      return
+    }
+    const { el, pipeline, range } = videoProperties
+    if (el === null || pipeline === undefined) {
+      return
+    }
+
+    // Extract range and update duration accordingly.
+    const [start = 0, end = duration] = range ?? [0, duration]
+    const __duration = duration ?? end ?? Infinity
+    setTotalDuration(__duration)
+
+    const updateProgress = () => {
+      const played = start + pipeline.currentTime
+      const buffered =
+        isHTMLMediaElement(el) && el.buffered.length > 0
+          ? start + el.buffered.end(el.buffered.length - 1)
+          : played
+      const total = __duration === Infinity ? buffered : __duration
+
+      const counter = `${Duration.fromMillis(played * 1000).toFormat(
+        'h:mm:ss',
+      )} / ${Duration.fromMillis(total * 1000).toFormat('h:mm:ss')}`
+      setProgress({
+        playedFraction: played / total,
+        bufferedFraction: buffered / total,
+        counter,
+      })
+    }
+    updateProgress()
+
+    // Use progress events on media elements
+    if (isHTMLMediaElement(el)) {
+      el.addEventListener('ended', updateProgress)
+      el.addEventListener('progress', updateProgress)
+      el.addEventListener('timeupdate', updateProgress)
+      return () => {
+        el.removeEventListener('timeupdate', updateProgress)
+        el.removeEventListener('progress', updateProgress)
+        el.removeEventListener('ended', updateProgress)
+      }
+    }
+
+    // Use polling when not a media element
+    const progressInterval = setInterval(updateProgress, 1000)
+    return () => {
+      clearInterval(progressInterval)
+    }
+  }, [videoProperties, duration, startTime, setTotalDuration])
+
+  const seek = useCallback(
+    (e: React.PointerEvent<HTMLDivElement>) => {
+      if (totalDuration === undefined) {
+        return
+      }
+
+      const { left, width } = e.currentTarget.getBoundingClientRect()
+      const fraction = (e.pageX - left) / width
+
+      onSeek(fraction * totalDuration)
+    },
+    [totalDuration, onSeek],
+  )
+
+  const [timestamp, setTimestamp] = useState({ left: 0, label: '' })
+  const __progressBarContainerRef = useRef<HTMLDivElement>(null)
+  useEffect(() => {
+    if (startTime !== undefined) {
+      __mediaTimeline.current.startDateTime = DateTime.fromISO(startTime)
+    }
+
+    const el = __progressBarContainerRef.current
+    if (el === null || totalDuration === undefined) {
+      return
+    }
+
+    const { left, width } = el.getBoundingClientRect()
```

Comment:
We should probably connect this to a resize event handler, and then use `left` and `width` in the dependency array for this hook (otherwise we rely on the timestamp to fire to handle resize).

Reply:
Should be future improvement if someone wants to implement proper resize behaviour.